cpp/src/arrow/io/file_test.cc

```diff
@@ -540,6 +544,9 @@ class TestPipeIO : public ::testing::Test {
 };
 
 TEST_F(TestPipeIO, TestWrite) {
+#ifdef __EMSCRIPTEN__
```

Comment:
Which should we use? `EMSCRIPTEN`? `__EMSCRIPTEN__`?

Reply:
I checked the docs and they only say `__EMSCRIPTEN__` so I've made it always be that.